zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RequestMapper.java

```diff
@@ -626,7 +626,11 @@ public static Authentication getAuthentication() {
         authenticationBuilder.tenants(
             authenticationContext.tenants().stream().map(TenantDTO::tenantId).toList());
 
-        authenticationBuilder.groupIds(authenticationContext.groups());
+        if (authenticationContext.groups() != null) {
+          authenticationBuilder.groupIds(authenticationContext.groups());
+          claims.put(Authorization.USER_GROUPS_CLAIMS, authenticationContext.groups());
+          claims.put(Authorization.GROUPS_CLAIM_ENABLED, true);
```

Comment:
❓ Could we keep this (and the code further down the chain) simpler and just use `USER_GROUPS_CLAIMS` then in the engine code we can just look to see if groups are present of not instead of having to keep two fields in step.